src/bundle.ts

```diff
@@ -17,10 +17,19 @@ export class Bundle {
 	public architecture: string;
 
 	/**
-	 * dockerfileHook: A function to be called with the resolved dockerfile
-	 * Note: The resolver will wait until the promise resolves before continuing
+	 * dockerfileHook: A function to be called with the
+	 * resolved dockerfile. If the hook returns a
+	 * string (or something which resolve to a string) we
+	 * replace the Dockerfile in place. Use this function for
+	 * further processing of the Dockerfile, after resolution
+	 * has determined the correct Dockerfile to use.
+	 *
+	 * NB: rather than null | string, we keep void | string to
+	 * avoid a breaking change
 	 */
-	private dockerfileHook: (content: string) => void | PromiseLike<void>;
+	private dockerfileHook: (
+		content: string,
+	) => void | string | PromiseLike<void | string>;
```

Comment:
A return type of `void |` makes no sense, as the other place I commented shows - please can you update to `undefined` which should be equivalent and non-breaking

Reply:
In case you choose to remove the `void` type, I was thinking that `Buffer` could be added as a potential performance improvement downstream (potentially avoiding a conversion to string just to satisfy the type information), given that `pack.entry()` accepts either string or Buffer:
```
private dockerfileHook: (
		content: string,
	) => undefined | string | Buffer | PromiseLike<undefined | string | Buffer>;
```